File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
       if(n == 1 || n == 7)
            return true;
        
        int temp = n;
        while(temp > 9){
            
            string s = to_string(temp);
            long long sm = 0;
            for(int i = 0; i<s.length(); i++){
                int rem = s[i]-'0';
                sm += rem * rem;
            }
            if(sm == 1 || sm == 7)
                return true;
            temp = sm;
        }
        return false;
    }
};

File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> st = new HashSet<>();

        while(true){

            int sm = 0;
            // sum of squares of digit
            while(n > 0){
                sm += Math.pow(n%10, 2);
                n = n/10;
            }

            if(sm == 1)
                return true;
            n = sm;
            if(st.contains(sm)){
                return false;
            }
            
            st.add(sm);
        }
        // return false;
    }
}